### DISCUSSION TITLE
llama3.2 - Error: cannot find tensor lm_head.weight (HF/Candle Rust)

### DISCUSSION THREAD

**Discussion by louis030195** [huggingface/candle#2514] [Q&A]
hi i'm trying to integrate this https://github.com/huggingface/candle/tree/main/candle-examples/examples/llama w llama323b in my code but getting

Error: cannot find tensor lm_head.weight

😦

the example works for me

i use `--features metal` (mac book pro m3 max 32 gb ram)

this is my code https://github.com/mediar-ai/screenpipe/blob/main/screenpipe-core/src/llama.rs

i just copy pasted the example 99% similar but somehow getting this error

any idea why?

**Comment by LaurentMazare** [comment]
Yes the new llama models share the lm-head with the embedding matrix. The github version of candle has been modified to handle this so you should probably update ot use it.
